Library/Homebrew/cmd/uses.rb

```diff
@@ -47,49 +47,48 @@ def uses
 
     uses = formulae.select do |f|
       used_formulae.all? do |ff|
-        if recursive
-          deps = f.recursive_dependencies do |dependent, dep|
-            if dep.recommended?
-              Dependency.prune if ignores.include?("recommended?") || dependent.build.without?(dep)
-            elsif dep.optional?
-              Dependency.prune if !includes.include?("optional?") && !dependent.build.with?(dep)
-            elsif dep.build?
-              Dependency.prune unless includes.include?("build?")
+        begin
+          if recursive
+            deps = f.recursive_dependencies do |dependent, dep|
+              if dep.recommended?
+                Dependency.prune if ignores.include?("recommended?") || dependent.build.without?(dep)
+              elsif dep.optional?
+                Dependency.prune if !includes.include?("optional?") && !dependent.build.with?(dep)
+              elsif dep.build?
+                Dependency.prune unless includes.include?("build?")
+              end
             end
-
-            # If a tap isn't installed, we can't find the dependencies of one
-            # its formulae, and an exception will be thrown if we try.
-            if dep.is_a?(TapDependency) && !dep.tap.installed?
-              Dependency.keep_but_prune_recursive_deps
+            reqs = f.recursive_requirements do |dependent, req|
```

Comment:
You could reraise if an environment variable is set or warn for developers only. 



> Remove the handler, but test brew uses against every formula in the official taps with every combination of arguments first (I'll happily run this)

Sounds like a good idea.